### PR TITLE
Links with unsafe URIs are escaped

### PR DESCRIPTION
**Pull request by YoranBrondsema**
This fixes https://github.com/bustlelabs/mobiledoc-kit/issues/525.

### CODE REVIEW

**Comment by YoranBrondsema** [comment]
@mixonic Would you be able to take a look when you have time? Thanks!

**Comment by gdub22** [comment]
I think this should live in user-land.  It can be added via a markup renderer.

**Comment by YoranBrondsema** [comment]
Should it really though? Security is something that affects all and wouldn't it be good if we can secure all users in one go against these kinds of XSS attacks? I agree that we can make it opt-out, but I don't see anything wrong with having a secure DOM renderer by default.

**Comment by YoranBrondsema** [comment]
Thanks @mixonic, that makes sense. I will adapt to escape rather than completely blocking the element.

**Comment by YoranBrondsema** [comment]
@mixonic I've been looking at [xss-filters](https://github.com/yahoo/xss-filters) to do the escaping, as I think it's better to use an external library for this kind of stuff to cover all the cases. Was that what you were thinking too?

If so, I've been trying to use [broccoli-multi-builder](https://github.com/bustlelabs/broccoli-multi-builder) to import the xss-filters NPM module. However, it fails with the error `Vendored module "xss-filters" missing "default" export` as xss-filters is not written in as an ES6 module. What do you propose?

Thanks in advance for your help!

**Comment by mixonic** [comment]
@YoranBrondsema I think we can use the method from glimmer:

* https://github.com/tildeio/glimmer/blob/8409f1e8b6eae6ca323aea45ad0d69ab9d05e586/packages/glimmer-runtime/lib/dom/sanitized-values.ts#L74
* https://github.com/emberjs/ember.js/blob/aeeddf7c00b42a340494465252481b5414d87009/packages/ember-glimmer/lib/protocol-for-url.js#L36

Which is basically to set the href on an A tag and check the protocol. @YoranBrondsema I'll ping you via bitter an perhaps we can pair to wrap it up.

File: tests/unit/utils/sanitization-utils-test.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    'www.google.com',
+    'tel:12345',
+    'mailto:john@doe.com'
+  ];
+
+  for (let i = 0; i < safe.length; i++) {
+    let url = safe[i];
+    assert.equal(sanitizeAttributeValue('href', url, 'a'), url);
+  }
+});
```
Comment (YoranBrondsema) [inline tests/unit/utils/sanitization-utils-test.js]: The tests for `reduceAndSanitizeAttributes` are missing.

**Comment by YoranBrondsema** [comment]
@mixonic Thanks for your help! Left a few small comments but I don't mind taking care of them. I'll do them tomorrow though as it's late now in Belgium! If you take care of the Node tests then I think we're good to go.

File: lib/renderers/0-3.js [hunk 2]
```diff
@@ -174,7 +177,7 @@ export default class Renderer {
         if (isValidMarkerType(tagName)) {
```
Comment (mixonic) [inline lib/renderers/0-3.js]: I'm not overly concerned as for 1.0 I'd like to drop the legacy 0.2 renderer. Probably once I ensure there is an ok path to upgrading existing documents though.

**Comment by YoranBrondsema** [comment]
Good to go from my side.

**Comment by YoranBrondsema** [comment]
@mixonic Any news about merging this PR? Thanks!

**Comment by mixonic** [comment]
@YoranBrondsema I'm going to merge for now. I'd like to know we haven't broken anything in Node before I cut a release though. My last work wrt this was spiking on a node runner for the qunit tests here. A smoke test in a fast boot running app would be ok for my purposes though.

**Comment by YoranBrondsema** [comment]
@mixonic Cool thanks. I played around with [qnit](https://github.com/andrasq/node-qunit) but it looks like the structure of the test files needs to be modified to:

````
module.exports = {
  'should run test': function(t) {
    t.printf("running test!\n");
    t.done();
  }
};
````

which I don't think is ideal.

If you have other suggestions I wouldn't mind looking into them though. Also I'd open to another pairing session to tackle this!

**Pull request merged**